generators/entity/templates/src/test/java/package/web/rest/_EntityResourceIntTest.java

```diff
@@ -195,7 +195,8 @@ public void setup() {
 
     @Before
     public void initTest() {<% if (databaseType == 'mongodb' || databaseType == 'cassandra') { %>
-        <%= entityInstance %>Repository.deleteAll();<% } %>
+        <%= entityInstance %>Repository.deleteAll();<% } if (searchEngine == 'elasticsearch') { %>
+        <%= entityInstance %>SearchRepository.deleteAll();<% } %>
```

Comment:
If the mapping of an entity changes, trying to insert an object into an old index can cause exceptions (ex. if a field type changes). Deleting the data is not enough, you need to delete the entity's index then recreate it with the new mapping:
```
        elasticsearchTemplate.deleteIndex(<%= entityClass %>.class);
        elasticsearchTemplate.createIndex(<%= entityClass %>.class);
        elasticsearchTemplate.putMapping(<%= entityClass %>.class);
```

This will also delete the data.